File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> freq; 
        // if (nums.size() == 0 ) {
        //     return false;
        // }
        int i = 0; 
        int j = nums.size() - 1; 
        while (i < j) {
            if (freq[nums[i]] > 0) {
                return true; 
            }
            freq[nums[i]]++;
            if (freq[nums[j]] > 0) {
                cout << freq[nums[j]]; 
                return true; 
            }
            // freq[nums[i]]++; 
            freq[nums[j]]++;
            cout << j; 
            cout << freq[2]; 

            i++; 
            j--; 
        }

        return false; 
        
    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        # s = HARJAS
        # K = 3
        if k >= len(s):
            return len(s)
        
        freq = [0] * 26
        start = 0
        max_l = 0
        max_freq  = 0
        #AAABABB
        #K = 1


        for i in range(len(s)):
            curr_length = i - start + 1
            index = ord(s[i]) - ord('A')
            print(index)
            freq[index] += 1
            for io in range(26):
                max_freq = max(max_freq, freq[io])    
            print("curr")
            print(max_freq)
            max_l = max(max_l, curr_length)
            if (curr_length - max_freq > k):
                max_l -= 1
            while curr_length - max_freq > k:
                index2 = ord(s[start]) - ord('A')
                freq[index2] -= 1
                start += 1
                curr_length = i - start + 1
        return max_l
        




        